src/conn/conn_api.c

```diff
@@ -1074,47 +1073,24 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
     WT_TRET(__wt_lsm_manager_destroy(session));
 
     /*
-     * After the async and LSM threads have exited, we shouldn't opening any more files.
+     * After the async and LSM threads have exited, we won't open more files for the application.
+     * However, the sweep server is still running and it can close file handles at the same time the
+     * final checkpoint is reviewing open data handles (forcing checkpoint to reopen handles). Shut
+     * down the sweep server and then flag the system should not open anything new.
      */
+    WT_TRET(__wt_sweep_destroy(session));
```

Comment:
@keithbostic  I've been sitting on the review of this because I'm nervous about this change. How confident are you that we don't need the sweep server to close handles after the checkpoint has finished? Maybe we can talk that through.

Reply:
Happy to talk it through: my initial pov is sweep only does operations based on a timer, that is, it can't be depended on to close any handle at any particular time, which means that if it is required by checkpoint, there's a bug in there by definition.

Reply:
You are right, and if it failed we would notice.